sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala

```diff
@@ -119,4 +113,55 @@ trait GpuPartitioning extends Partitioning with Arm {
       sliceRange.close()
     }
   }
+
+  /**
+   * Compress contiguous tables representing the splits into compressed columnar batches.
+   * Contiguous tables corresponding to splits with no data will not be compressed.
+   * @param outputBatches where to collect the corresponding columnar batches for the splits
+   * @param codec compression codec to use
+   * @param contiguousTables contiguous tables to compress
+   */
+  def compressSplits(
+      outputBatches: ArrayBuffer[ColumnarBatch],
+      codec: TableCompressionCodec,
+      contiguousTables: Array[ContiguousTable]): Unit = {
+    withResource(codec.createBatchCompressor(maxCompressionBatchSize)) { compressor =>
+      // tracks batches with no data and the corresponding output index for the batch
+      val emptyBatches = new ArrayBuffer[(ColumnarBatch, Int)]
+
+      // add each table either to the batch to be compressed or to the empty batch tracker
+      contiguousTables.zipWithIndex.foreach { case (ct, i) =>
+        if (ct.getTable.getRowCount == 0) {
```

Comment:
Should I assume that tables with only rows and no columns have already been filtered out, because these are `ContiguousTable` instances? I'm not sure it matters because I don't think we hit that in practice outside of broadcast.

Reply:
Yes, we cannot build a `Table` instance with no columns.